### ISSUE TITLE
Game data files (*.POO) are missing!

### ISSUE THREAD

**Issue by marvin-SL** [jakegadget/project-edn-server#4]
Hello, I followed the guide to generate the config folder and cthen copy to `project-edn-server/Config`, but when i run the `dotnet run` command i've got the error `Game data files (*.POO) are missing!`. I looked into the code and it looking for `.csv` files when i've only got encrypted `.poo` files.  
here https://github.com/yes-but-also-no/project-edn-server/blob/e496c3208bbab5615c423cecc449b3dcf0708449/GameServer/Configuration/PooReader.cs#L40

and here https://github.com/yes-but-also-no/project-edn-server/blob/e496c3208bbab5615c423cecc449b3dcf0708449/GameServer/Configuration/PooReader.cs#L161

Did I miss something ? 

thanks

**Comment by yes-but-also-no** [comment]
Hi Marvin,

Thanks for reaching out. In an effort to not get on the wrong side of anyone's copyrighted content, the repository only contains original creations, not the game files. 

It looks like you are on the right track with that, since you have the game's encrypted .poo files already. There is another repository that contains scripts for decrypting the .poo files, here is a link:

[https://github.com/yes-but-also-no/project-edn-tools](https://github.com/yes-but-also-no/project-edn-tools)

If you are just looking to run the server, you can skip the patching process section and head right down to the "server file extraction" instructions. 

Basically, you will just clone that repository, drag your game install onto the *SERVER-extract-maps-poo.bat* script, and it will dump a folder with all of the maps and poo files decrypted. 

Feel free to update this issue if you are still having issues and I will try to point you in the right direction.

Thanks!

**Comment by marvin-SL** [comment]
Thanks for your very quick answer ! I've followed this guide, this is how I generated the Config folder. But the POO files are encrypted even after used the batch `SERVER-extract-maps-poo.bat`. 
But it seems I should have decrypted `.csv` files instead of encrypted `poo` files, is that correct ? Maybe i did something wrong and this is why my files are encrypted  `.poo`
"- Decrypt the *.poo files and rename them to .csv" Yes this is correct, I'll try again tomorrow. Thank you very much

**Comment by yes-but-also-no** [comment]
No problem. Did the script print any errors while running, or did it report successfully?

The tools were thrown together fairly quickly, so I know a few people have had issues here and there depending on their version of windows, language settings, and path names.

The extract maps script is actually very simple, so I would open up that script file and see if you need to make any modifications. You could also just manually decrypt them. All the script does, in essence, is run l2encdec.exe on each poo file with the -t and -l flags. If it is unable to decrypt, it runs it again using the -t and -d flags, in the event the game files had already been decrypted and re-encrypted in the past. Then it just renames them to .csv.

If you reach out to me on discord, I can try and help you further.

**Comment by marvin-SL** [comment]
Ah I found my mistake : I put the script SERVER-extract-maps-poo.bat in the `project-edn-v1.4` folder, and then drag n drop the Exteel folder to it, but it needed  l2encdec.exe  to work (I had an error about a missing path in the SERVER-extract-maps-poo.bat), instead of simply DnD the Exteel folder into `project-edn-server-master/SERVER-extract-maps-poo.bat ` Now my poo files are decrypted :) . I'll do the next steps tomorrow. 
Thank you for you awesome work, me and my friend were looking for a project like this since NCSoft closed the game ! 👍

**Comment by yes-but-also-no** [comment]
No problem! Glad to help! I haven't really been maintaining the project recently due to other things taking priority, so its great to have other people pitching in!
